src/library/basesqltablemodel.cpp

```diff
@@ -633,13 +622,18 @@ int BaseSqlTableModel::fieldIndex(const QString& fieldName) const {
         // column or a source column.
         int sourceTableIndex = m_trackSource->fieldIndex(fieldName);
         if (sourceTableIndex > -1) {
-            // Subtract one from the fieldIndex() result to account for the id column
+            // Subtract one from the fieldIndex() because the id column is in both
             return m_tableColumns.size() + sourceTableIndex - 1;
         }
     }
     return tableIndex;
 }
 
+int BaseSqlTableModel::maxFieldIndex() const {
+    // Subtract one to remove the id column which is in both
+    return m_tableColumns.size() + (m_trackSource ? m_trackSource->maxFieldIndex() : 0);
```

Comment:
That doesn't look correct to me.

The brittle _max index_ = _size minus one_ semantics are error prone.

`maxFieldIndex()` should better become `endFieldIndex()` which would be more consistent with how ranges are represented in C++.